### ISSUE TITLE
feat(qwen3_5): load fine-grained FP8 checkpoints by requantizing 128x128 block scales to mxfp8

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1361]
## Summary

The vendor FP8 releases of the Qwen3.5 family (`Qwen/Qwen3.8-27B-FP8` and siblings) store every projection as `E4M3` bytes in `*.weight` plus a `bfloat16` inverse scale per 128x128 block in `*.weight_scale_inv`, declared by `quantization_config: {"quant_method": "fp8", "fmt": "e4m3", "weight_block_size": [128, 128]}`. mlxcel's loader decodes `F8_E4M3` tensors to f16 values but never multiplies by the block scale and never reads `weight_scale_inv`, so such a checkpoint either fails on the unexpected sidecar or loads a weight that is off by the per-block scale. `docs/supported-models.md` records the gap. This change reconstructs each block-scaled tensor at load and requantizes it to MLX's native `mxfp8` layout (`group_size 32`, `bits 8`), which the quantized-linear path already executes.

## Current behavior

`src/models/sanitize.rs:1116-1140`: `SafeTensorDtype::F8_E4M3` is decoded element-wise with `f8_e4m3_to_f32` and cast to f16, with no scale applied:

```rust
SafeTensorDtype::F8_E4M3 => {
    // MLX has no native float8 dtype; convert F8_E4M3 → f16 at load time.
    // Used by nvfp4 Gemma 4 checkpoints (weight_scale tensors).
```

No loader consumes a `weight_scale_inv` tensor; the only mention drops it (`src/models/mistral4.rs:908-912`, "Remove weight_scale_inv / activation_scale keys (FP8 quantization artifacts)"). `docs/supported-models.md:97` (Qwen3.5-VL / Qwen3.8 entry): "The FP8 and ModelOpt NVFP4 builds of the same model are out of scope: they use vLLM-style `weight_scale_inv` block scales that mlxcel cannot read."

The pieces needed already exist: `mlxcel_core::quantize_weights_with_mode(w, group_size, bits, mode)` (`src/lib/mlxcel-core/src/lib.rs:2674-2679`), `mxfp8` in `SUPPORTED_QUANTIZATION_MODES` (`src/lib/mlxcel-core/src/layers.rs:1325`) and in the `UnifiedLinear` mode handling (`src/lib/mlxcel-core/src/layers.rs:242-301`), per-tensor quantization inference in `UnifiedLinear::from_weights`, and the Qwen3.5 sanitize entry points (`src/models/qwen3_5.rs:2898` `sanitize_weights`, `:3140` `sanitize_moe_weights`).

## Expected behavior

Detection: `quantization_config.quant_method == "fp8"`, `fmt` absent or `"e4m3"`, `weight_block_size == [128, 128]`, on `model_type` `qwen3_5` / `qwen3_5_moe` (text or VLM wrapper). Any other `weight_block_size` is a named load error. `Qwen/Qwen3.8-27B-FP8` declares exactly this (`activation_scheme: "dynamic"`, `fmt: "e4m3"`, `weight_block_size: [128, 128]`) and lists the whole vision tower (`model.visual.*`) plus the norms, routers and `mtp.fc` under `modules_to_not_convert`, so those tensors arrive as plain bf16 and take the existing dense path; only tensors with a `weight_scale_inv` sibling are converted, which makes the `modules_to_not_convert` list informational.

Per tensor pair `(name.weight: uint8 [R, C] E4M3 bytes, name.weight_scale_inv: bf16 [ceil(R/128), ceil(C/128)])`:

```
decoded      = e4m3_to_bf16(weight)                                  # [R, C]
padded       = pad(decoded, rows to R' = ceil(R/128)*128, cols to C' = ceil(C/128)*128, 0)
blocks       = padded.reshape(R'/128, 128, C'/128, 128)
restored     = (blocks * scale_inv[:, None, :, None]).reshape(R', C')[:R, :C]
(q, scales)  = quantize(restored, group_size = 32, bits = 8, mode = "mxfp8")   # no biases
```

`name.weight` is replaced by `q`, `name.weight_scale_inv` is removed, `name.scales` is added, and the model loads with the effective config `quantization: {"group_size": 32, "bits": 8, "mode": "mxfp8"}` applied to every tensor that carried a block scale; tensors without a sidecar (norms, embeddings, `conv1d`, `A_log`, `dt_bias`) stay dense. `C % 32 == 0` is required (all Qwen3.5 projections satisfy it); otherwise a named load error. The forward pass then runs on the existing `mxfp8` quantized-matmul path with no per-token dequantization.

Peak memory during conversion must stay bounded: convert one tensor at a time and `eval` the quantized pair before moving on, so the bf16 `restored` copy (4x the E4M3 bytes) is transient.

## Implementation plan

1. `src/models/config.rs` (`Quantization` / quantization-config readers): add `pub fn qwen_fp8_block_quantization(full_config: &Value) -> Result<Option<Quantization>, String>` returning `Some(Quantization { group_size: 32, bits: 8, mode: "mxfp8" })` for the detection rule above and `Err` for an unsupported block size.
2. `src/models/sanitize.rs`: add `pub fn requantize_block_fp8_weights(weights: WeightMap, block: usize) -> Result<WeightMap, String>` implementing the per-tensor procedure with `mlxcel_core` ops (`from_bytes` as `UINT8`, an `e4m3_to_bf16` decode via the existing `f8_e4m3_to_f32` table lifted into an MLX `take` over a 256-entry LUT so the decode runs on device, `pad`, `reshape`, `multiply`, `slice`, `quantize_weights_with_mode(..., 32, 8, "mxfp8")`). Keep the E4M3 bytes raw for these tensors by loading `*.weight` as `UINT8` when a `*.weight_scale_inv` sibling exists (a `SelectiveLoadMode` branch next to `src/models/sanitize.rs:1116`), since the current f16 decode loses the pairing.
3. `src/loading/special.rs:115-129` (`SpecialWeightLoaderKind::Qwen35`) and `src/loading/vlm_qwen.rs:353` (`load_qwen3_5_vlm_with_variant`): when step 1 returns `Some`, run step 2 before `sanitize_moe_weights` / `sanitize_weights`, and inject the effective `quantization` into `text_config` the same way `qwen35_text_config` (`src/loading/special.rs:44-58`) merges the top-level block today. The MTP drafter loader (`src/models/qwen3_5_mtp_target.rs`) takes the same pre-pass if the checkpoint bundles `mtp.*` FP8 tensors.
4. `mlxcel list` / `docs/supported-models.md:97`: replace the out-of-scope sentence with the supported FP8 layout and its memory cost (mxfp8 is 8 bits per weight plus one E8M0 scale per 32 values).
5. `Cargo` / CI: no new dependency; the LUT decode is pure MLX.

## Validation

(a) Unit tests in `src/models/sanitize_tests.rs` (or the existing `#[cfg(test)]` block in `sanitize.rs`):

- `fp8_block_requantize_matches_direct_path`: random bf16 `[130, 160]` weight, quantize to E4M3 bytes with `f32_to_f8_e4m3`, random `[2, 2]` bf16 `scale_inv`; assert `requantize_block_fp8_weights` output equals `quantize_weights_with_mode(decode(bytes) * expanded_scales, 32, 8, "mxfp8")` bitwise (both `w` and `scales`), and that dequantizing the result reproduces `restored` within the mxfp8 tolerance (max abs error <= 1/16 of each group's max).
- `fp8_block_requantize_rejects_odd_width`: `C = 100` returns `Err`.
- `qwen_fp8_config_detection`: `[128, 128]` gives `Some(mxfp8 32/8)`, `[64, 64]` gives `Err`, `quant_method: "int4"` gives `None`.

(b) Real checkpoint: `Qwen/Qwen3.8-27B-FP8` (vendor FP8 release of the validated `mlx-community/Qwen3.8-27B-4bit` architecture; about 28 GB of E4M3 weights).

```
./target/release/mlxcel generate -m models/Qwen3.8-27B-FP8 -p "Write a limerick about a Rust compiler." -n 64
```

Acceptance: finite, fluent output; top-5 logits of the first generated position agree with the bf16 original `Qwen/Qwen3.8-27B` (loaded through the existing bf16 path, with `MLXCEL_FUSED_MOE=0` if the MoE sibling is used) within the mxfp8 tolerance, and greedy output agrees for at least the first 16 tokens on a 64-token prompt. Peak RSS during load stays below 1.5x the final resident size.

## Acceptance criteria

- [x] `quant_method: "fp8"` with `weight_block_size [128, 128]` is detected on `qwen3_5` / `qwen3_5_moe`; other block sizes are named errors
- [x] Block-scaled tensors are reconstructed and requantized to `mxfp8` 32/8 at load, one tensor at a time
- [ ] `Qwen/Qwen3.8-27B-FP8` loads and generates; logits agree with the bf16 original within the mxfp8 tolerance (loads and generates: verified on GB10 / CUDA with `MLX_ENABLE_TF32=0`, 23.36 GB resident and 24.28 GB peak during load, fluent 207-token answer. Logit parity against the bf16 original: unverified, because `Qwen/Qwen3.8-27B` (55.6 GB) was deliberately not fetched on the validation host, so the criterion stays unticked.)
- [x] docs/supported-models.md updated
- [ ] cargo test --workspace --profile test-fast --features metal,accelerate passes (not runnable on this host: Linux / aarch64 / CUDA, no Metal and no Accelerate. Unrun. The CUDA equivalent was run for the changed scope instead.)
- [x] cargo clippy --workspace --all-targets -- -D warnings and cargo fmt --all -- --check pass

## Out of scope

- ModelOpt NVFP4 builds of Qwen3.8 (different sidecar layout, `weight_scale` / `weight_scale_2`).
- Other families' FP8 releases (DeepSeek V3 UE8M0, Mistral 4); they can reuse `requantize_block_fp8_weights` once their block sizes are confirmed.
